dask/dataframe/tests/test_io.py

```diff
@@ -57,7 +68,7 @@ def test_read_csv(open_comp_pair, infer):
         result = f.compute(get=dask.get)
         # index may be different
         assert eq(result.reset_index(drop=True),
-                  pd.read_csv(fn, compression=compression, lineterminator='\n'))
+                  pd.read_csv(io.StringIO(text), lineterminator='\n'))
```

Comment:
Sorry, I should have been more specific. If you look at the test failures, you can see that the python 2.7 failures are because you're passing a `str` to `io.StringIO`, which expects `unicode`. The python 3 failures are due to `pd.read_csv` being called with `compression='xz'`, which isn't implemented in pandas 0.18.0.

How I would fix these problems is by:
1. Reverting the change above, or replacing `text` with `text.encode()` (I'd probably do the first one)
2. Skipping the `xz` compression if pandas version isn't greater than 0.18.0.